dask/dataframe/groupby.py

```diff
@@ -1132,8 +1139,8 @@ def __init__(
         slice=None,
         group_keys=GROUP_KEYS_DEFAULT,
         dropna=None,
-        sort=None,
-        observed=None,
+        sort=True,
```

Comment:
Doesn't `SORT_SPLIT_OUT_WARNING` imply that we are not actually making this change yet?

Reply:
I was a bit cheeky here in that I changed it for the private `_Groupby` class, but it still defaults to `None` in `DataFrame.groupby`, so all users should see the same defaults. Happy to switch this back too if you think it's confusing.

Reply:
Ah, that makes sense. Sorry for missing that. No need to change this back, as long as the change isn't "breaking" just yet.